### PR TITLE
fix z-index action to account for deleted elems

### PR DESCRIPTION
**Pull request by dwelle**
fixes https://github.com/excalidraw/excalidraw/issues/989

Implementation follows @vjeux [suggestion](https://github.com/excalidraw/excalidraw/issues/1069#issuecomment-603064630) of auto-selecting neighboring deleted elements.

Comments & tests tomorrow, and merge then. Pushed as RFC.

### CODE REVIEW

File: src/scene/selection.ts [hunk 1]
```diff
@@ -49,19 +49,6 @@ export function deleteSelectedElements(
...
-) {
-  const selectedIndices: number[] = [];
-  elements.forEach((element, index) => {
-    if (appState.selectedElementIds[element.id]) {
-      selectedIndices.push(index);
-    }
-  });
-  return selectedIndices;
-}
-
```
Comment (dwelle) [inline src/scene/selection.ts]: Removed because it was used only by z-index manip and no longer used as of now.

**Comment by petehunt** [comment]
just tested it, seems to work well

**Comment by dwelle** [comment]
Oh yea, gotta wrap this up.

**Comment by dwelle** [comment]
Got slowed down on tests.. gonna be pushing in a tiny bit.

**Comment by dwelle** [comment]
Ready for review. Got stuck on debugging a test issue that was caused by a typo.

**Comment by vjeux** [comment]
Unit tests are great to test algorithms and edge cases. I feel like this is one of those cases. End to end tests are more useful for sanity check that things are working and not completely broken.

Having both make sense to me.
